src/machine/machine_nrf52xxx.go

```diff
@@ -324,6 +338,16 @@ func (spi SPI) Tx(w, r []byte) error {
 	return nil
 }
 
+// Read implements [io.Reader]. And reads as many bytes as the given buffer is long
+func (spi *SPI) Read(r []byte) (int, error) {
+	return len(r), spi.Tx(nil, r)
+}
+
+// Write implements [io.Writer]. And writes as long as there are bytes in w.
+func (spi *SPI) Write(w []byte) (int, error) {
+	return len(w), spi.Tx(w, nil)
+}
+
```

Comment:
Read and Write are not part of the SPI interface. It is easy enough to implement an abstraction if needed that works for all SPI hardwares of all MCUs

```go
func writeSPI(w []byte, spi interface {Tx(w,r []byte) error}) (int,error) {
    return len(w), spi.Tx(w,nil)
}
```

Reply:
Yeah, lets do it! Actually I was surprised that SPI is not supplying it, since it would have seemed the more natural way to interface with SPI than the current one.